mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java

```diff
@@ -61,25 +62,76 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.github.fge.lambdas.Throwing;
 import com.github.steveash.guavate.Guavate;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableSet;
 import com.google.common.collect.Sets;
-import com.google.common.collect.Sets.SetView;
 
 public class StoreMessageIdManager implements MessageIdManager {
+
+    private static class MessageMoves {
+        private final ImmutableSet<MailboxId> previousMailboxIds;
+        private final ImmutableSet<MailboxId> targetMailboxIds;
+
+        MessageMoves(Collection<MailboxId> previousMailboxIds, Collection<MailboxId> targetMailboxIds) {
+            this.previousMailboxIds = ImmutableSet.copyOf(previousMailboxIds);
+            this.targetMailboxIds = ImmutableSet.copyOf(targetMailboxIds);
+        }
+
+        boolean isChange() {
+            return !previousMailboxIds.equals(targetMailboxIds);
+        }
+
+        Set<MailboxId> addedMailboxIds() {
+            return Sets.difference(targetMailboxIds, previousMailboxIds);
+        }
+
+        Set<MailboxId> removedMailboxIds() {
+            return Sets.difference(previousMailboxIds, targetMailboxIds);
+        }
+    }
+
+    private static class MetadataWithMailboxId {
+        private final MessageMetaData messageMetaData;
+        private final MailboxId mailboxId;
+
+        public MetadataWithMailboxId(MessageMetaData messageMetaData, MailboxId mailboxId) {
+            this.messageMetaData = messageMetaData;
+            this.mailboxId = mailboxId;
+        }
+    }
+
+    private static class WrappedException extends RuntimeException {
```

Comment:
we can replace this by Throwing functions

Reply:
It's just a class move, nothing to do with my code.

+ I don't see how Throwing functions coul help here....